### DISCUSSION TITLE
Why is `RandomLinkSplit` on an undirected graph returning directed edge label index?

### DISCUSSION THREAD

**Discussion by saiden89** [pyg-team/pytorch_geometric#5169] [Q&A] [answered]
Hi everyone. Recently I've been dealing a lot with GNN link prediction models, and one of the very first steps to do is to partition the data into train, validation and testing; you know, the usual stuff.
To my understanding, in PyG this is achieved through `RandomLinkSplit`:
```python3
from torch_geometric.datasets import FakeDataset
from torch_geometric.transforms import RandomLinkSplit
from torch_geometric.utils import is_undirected


dataset = FakeDataset(
    num_graphs=1, 
    avg_num_nodes=100, 
    num_node_types=1, 
    edge_dim=20, 
    is_undirected=True
)


transform = RandomLinkSplit(is_undirected=True)
splits = transform(dataset[0])
```

What I'm currently not grasping is the reason beneath providing the supervision edge index (`edge_label_index`) in *directed* form, even when specifically informing the transform that we are currently dealing with a undirected network.

```python3
print(is_undirected(splits[0].edge_index)) # True
print(is_undirected(splits[0].edge_label_index)) # False
```
Is this intended behaviour? If so, what am I currently missing?

**Comment by rusty1s** [comment, marked as answer]
This is intended behavior since an undirected `edge_label_index` would just blow up the amounts of labels artificially since one usually uses a symmetric encoder to predict the links in this scenario. If you still want to incorporate both directions as part of your labels, you can simply flip the matrix and use it for supervision as well. WDYT?

**Reply by saiden89** [reply]
I understand the reasoning, thanks, but I don't think it's so far-fetched the idea that someone would use an asymmetric decoder (say, an MLP) to predict the edges based on two node's concatenated emeddings.
From a user perspective (especially from a novice) I think there is an expectation that such major conversions should not pass silently and with no documentation to support the decision.
I you want I'd be happy to help in this regard :).

**Reply by rusty1s** [reply]
I would very much appreciate help here. We can update the doc-string for sure. If you think this is absolutely necessary, we can also think about adding an argument to return undirected edge labels.

**Reply by saiden89** [reply]
I can submit a PR in the following days for updating the docstring. Regarding feature enhancement, is there a way to check where community and developer consensus lies (open an issue?). I'm kind on the fence, on one hand more features are often nice, on the other hand I wouldn't want to bloat the codebase with useless features.

**Reply by rusty1s** [reply]
We can create an issue around this and try to get people's opinion in.

**Reply by saiden89** [reply]
Discussion continues on #5190.
